matching/dictionary.go

```diff
@@ -12,19 +14,21 @@ type dictionaryMatch struct {
 
 func (dm dictionaryMatch) Matches(password string) []*match.Match {
 	var results []*match.Match
-	pwLower := strings.ToLower(password)
+	pwLower := []rune(strings.ToLower(password))
+
+	passwordRunes := []rune(password)
 
 	for dictionaryName, rankedDict := range dm.rankedDictionaries {
-		for i := range password {
-			for delta := range password[i:] {
+		for i := range passwordRunes {
+			for delta := range passwordRunes[i:] {
 				j := i + delta
-				word := pwLower[i : j+1]
+				word := string(pwLower[i : j+1])
 				if val, ok := rankedDict[word]; ok {
 					matchDic := &match.Match{
 						Pattern:        "dictionary",
 						I:              i,
 						J:              j,
-						Token:          password[i : j+1],
+						Token:          string(passwordRunes[i : j+1]),
```

Comment:
It looks like this change will give different I,J indexes for strings containing unicode characters than before...
Is this why you had to change testdata/output.json ? If so, could we instead find a fix that doesn't change the previous meaning of I,J as indexes into the original password string ?

Reply:
I _think_ that the original test output may have been using the ascii version of that character but in golang it's utf8 proper? I don't have node setup to regenerate output.json: maybe worth seeing if the change I mad to gen.sh influences output.json?

What is clear is that strings.ToLower can give you a string of length less than that of the input. So indexing into pwLower with indices from iterating on password can go out of bounds. Perhaps the ASCII substrings (I'm assuming the dictionaries are probably printable ASCII) should be extracted and those converted into lower case for the purposes of comparison?

Reply:
Can you try to only change line 21 in the original version from 
```
word := pwLower[i : j+1]
```
to
```
word := strings.ToLower(password[i : j+1])
```

I think this should solve the problem while keeping the previous index semantic